src/car/car.service.ts

```diff
@@ -52,6 +48,54 @@ export class CarService {
         return carUsage;
     }
 
+    async getAllCarsUsage(): Promise<object> {
+        const monthName = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+        //! VERY IMPORTANT QUERY - fully check it out
+        const query = `SELECT cars.car_id, reservations.id reservation_id,car_name, car_license_plate, total_price, start_date, end_date FROM cars LEFT JOIN reservations ON cars.car_id = reservations.car_id`;
+
+        const carsData = await this.databaseService.executeQuery(query);
+
+        const allCarsUsage = carsData.rows.reduce((acc, car) => {
+            const startDate = new Date(car.start_date);
+            const endDate = new Date(car.end_date);
+            const days = Math.ceil((endDate.getTime() - startDate.getTime()) / (1000 * 3600 * 24));
+            const month = startDate.getMonth();
+
+            if (acc[car.car_id]) {
+                if (acc[car.car_id].record[month]) {
+                    acc[car.car_id].record[month].days += days;
+                    acc[car.car_id].record[month].count++;
+                } else {
+                    acc[car.car_id].record[month] = { days, count: 1 };
+                }
+            } else {
+                acc[car.car_id] = { record: { [month]: { days, count: 1 } }, license_plate: car.car_license_plate, name: car.car_name };
+            }
+            return acc;
+        }, {});
+
+        const allCarsUsageUpdated = {};
+
+        for (const car in allCarsUsage) {
+            allCarsUsageUpdated[car] = {};
+            allCarsUsageUpdated[car].record = {};
+            for (const month in allCarsUsage[car].record) {
+                allCarsUsage[car].record[month].percentage = ((allCarsUsage[car].record[month].days / 30) * 100).toFixed(2);
+
+                allCarsUsageUpdated[car].record[monthName[month]] = allCarsUsage[car].record[month];
+                allCarsUsageUpdated[car] = { ...allCarsUsageUpdated[car], license_plate: allCarsUsage[car].license_plate, name: allCarsUsage[car].name };
+
+                if (allCarsUsageUpdated[car].record[monthName[month]].days === 0) {
+                    allCarsUsageUpdated[car]['record'] = {};
+                    delete allCarsUsageUpdated[car].record[monthName[month]];
+                }
```

Comment:
why do you delete it? it's okay if usage = 0%

Reply:
Because the "count" property will show 1 even if the usage is 0%. Should I remove this section as leave it as is? 
`
"3": {
        "record": {
            "January": {
                "days": 0,
                "count": 1,
                "percentage": "0.00"
            }
        },
        "license_plate": "AA111AA",
        "name": "Toyota"
    }
`

Reply:
what is count? you can just remove it

Reply:
The count shows how many times a car has been rented. for instance: 
`{ "January": { "days": 18, "count": 3, "percentage": "..." }`
means on January a car is rented 3 times resulting in 18 days total.
I can remove it if it has no use. 
And beside the "count" property, the month name of January for the cars with no rental records bugged me , so I decided to return an empty record property.